cmd/argocd/commands/login.go

```diff
@@ -112,6 +112,8 @@ argocd login cd.argoproj.io --core`,
 				ServerName:           globalClientOpts.ServerName,
 			}
 
+			ctxName = globalClientOpts.Context
```

Comment:
```suggestion
			ctxName = globalClientOpts.Context
```

why is this required? Shouldn't we get the `ctxName` value automatically from the flag itself?

Reply:
That global option is the flag where it's being read from. If it's not supplied then it's created.

Reply:
@nitishfy So today if you run `argocd login argocd.my.host --sso --argocd-context my-argocd` the context that gets created is called `argocd.my.host` and the option is ignored. The code path today sets the `ctxName` variable equal to the server name if its unset (which is always is). Then it performs some additional transformations on the `ctxName` variable. What I did was just default the `ctxName` to the command line flag in this change.